### PR TITLE
`Feat/events` --> `quality-assurance`

### PR DESCRIPTION
**Pull request by Jooty**
## Changelog

- Added a bunch of missing type structures not serialized with the client
- Added `WizardZoneTrigger`.
- Added `WizardZoneTriggerSupervisor`
- Added `ResultDispatcher`
- Added players to post event `EnterZone` upon zone entrance
- Added trigger cooldowns

## Writings

Changes the way we think about zone "volumes" and "triggers." To be clear, zone volumes are 3D areas inside of a zone that may trigger a "event" if a player were to enter or leave. Our events are also referred to as "triggers," and a `WizardZone` is in charge of monitoring them. Not all events may be posted by a volume, but all volumes post an event.

In short: a player enters a _volume_, posting an _event_ to the `WizardZone` to activate a _trigger_.

Another objective of this PR is to make posting an event generic. As a volume posts an event to the `WizardZone`, completing a quest goal may also post an event to the same `WizardZone`, with the sender as context.

A trigger has a number of requirements (which are not included in this PR) and results. These results include a number of dynamic changes, such as changing the player's health or teleporting them. The `ResultDispatcher` serves to handle any of the expected results.

### CODE REVIEW

**Review by gmrush — APPROVED** [review]
LGTM B-)
